diff --git a/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf b/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
index c531dbb..7fd075a 100644
--- a/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
+++ b/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
@@ -8,7 +8,8 @@ require conf/machine/include/machine-xilinx-default.inc
 MACHINE_FEATURES = "ext2 vfat usbhost"
 
 # u-boot configuration
-PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
+#PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
+PREFERRED_PROVIDER_virtual/bootloader ??= "u-boot-xlnx"
 UBOOT_MACHINE = "xilinx_zynq_virt_defconfig"
 SPL_BINARY ?= "spl/boot.bin"
 UBOOT_ELF = "u-boot"
diff --git a/meta-xilinx-bsp/recipes-bsp/platform-init/platform-init.bb b/meta-xilinx-bsp/recipes-bsp/platform-init/platform-init.bb
index 32509b0..31c6d08 100644
--- a/meta-xilinx-bsp/recipes-bsp/platform-init/platform-init.bb
+++ b/meta-xilinx-bsp/recipes-bsp/platform-init/platform-init.bb
@@ -11,6 +11,7 @@ inherit xilinx-platform-init
 
 COMPATIBLE_MACHINE = "$^"
 COMPATIBLE_MACHINE_picozed-zynq7 = "picozed-zynq7"
+COMPATIBLE_MACHINE_microzed-zynq7 = "microzed-zynq7"
 
 LICENSE = "GPLv2+"
 LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/GPL-2.0;md5=801f80980d171dd6425610833a22dbe6"
