diff --git a/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf b/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
index c531dbb..7fd075a 100644
--- a/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
+++ b/meta-xilinx-bsp/conf/machine/microzed-zynq7.conf
@@ -8,7 +8,8 @@ require conf/machine/include/machine-xilinx-default.inc
 MACHINE_FEATURES = "ext2 vfat usbhost"
 
 # u-boot configuration
-PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
+#PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
+PREFERRED_PROVIDER_virtual/bootloader ??= "u-boot-xlnx"
 UBOOT_MACHINE = "xilinx_zynq_virt_defconfig"
 SPL_BINARY ?= "spl/boot.bin"
 UBOOT_ELF = "u-boot"
